include/openmc/capi.h

```diff
@@ -118,6 +118,7 @@ int openmc_regular_mesh_get_params(
 int openmc_regular_mesh_set_dimension(int32_t index, int n, const int* dims);
 int openmc_regular_mesh_set_params(int32_t index, int n, const double* ll,
   const double* ur, const double* width);
+int openmc_remove_tally(int32_t id);
```

Comment:
All of our C API functions related to tallies pretty much uniformly take the _index_ as an argument rather than the id, so I would propose we change this to take the index as well. If you only know the id, you can use `openmc_tally_get_id` to determine the index.

Reply:
Sounds good! I will fix this

Reply:
For this though, I currently have 
```
  // check that id is in the map
  if (model::tally_map.count(id)!=1) {
    return OPENMC_E_INVALID_ID;
  }
```
If I am given an index instead, does this mean I'll just compare the `index` to the size of the tallies vector? If the `index` is greater than the size, the tally does not exist. Should I still return `OPENMC_E_INVALID_ID` (since technically it is an invalid index) or is there a more appropriate return type?

Reply:
Take a look at any of the other `openmc_tally_*` functions in tally.cpp as they all have consistent logic for treating this

Reply:
> If I am given an index instead, does this mean I'll just compare the index to the size of the tallies vector?

Yeah I would do a similar check as other functions that take an index into global vectors. Something like:

`if (index >= 0 && index < model::tallies.size()) { ... }`

>  Should I still return OPENMC_E_INVALID_ID (since technically it is an invalid index) or is there a more appropriate return type?

I'd say that returning `OPENMC_E_OUT_OF_BOUNDS` is the way to go here.